client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import {
+  keys,
+  find
+} from 'min-dash';
+
+import { getEngineProfile } from '../../../util/parse';
+
+const BPMN_TAB_TYPE = 'bpmn';
+const CLOUD_BPMN_TAB_TYPE = 'cloud-bpmn';
+const DMN_TAB_TYPE = 'dmn';
+const CLOUD_DMN_TAB_TYPE = 'cloud-dmn';
+
+const DIAGRAM_BY_TAB_TYPE = {
+  'bpmn': [ BPMN_TAB_TYPE, CLOUD_BPMN_TAB_TYPE ],
+  'dmn': [ DMN_TAB_TYPE, CLOUD_DMN_TAB_TYPE ]
+};
+
+export default class DeploymentEventHandler {
+  constructor(props) {
+
+    const {
+      subscribe,
+      track
+    } = props;
+
+    this.track = track;
+
+    this.subscribeToDeploymentEvents(subscribe);
+  }
+
+    subscribeToDeploymentEvents = (subscribe) => {
+
+      subscribe('deployment.done', async (event) => {
+        const { tab } = event;
+        const type = getDiagramType(tab.type);
+
+        await this.trackDeploymentAction(type, true, event);
+      });
+
+      subscribe('deployment.error', async (event) => {
+        const { tab } = event;
+        const type = getDiagramType(tab.type);
+
+        await this.trackDeploymentAction(type, false, event);
+      });
+
+    }
+
+    trackDeploymentAction = async (diagramType, success, event) => {
+      const {
+        context,
+        deployedTo,
+        targetType,
+        error,
+        tab
+      } = event;
+
+      const {
+        file
+      } = tab;
+
+      const {
+        contents
+      } = file;
+
+      if (!diagramType) {
+        return;
+      }
+
+      const baseEvent = context === 'deploymentTool' ? 'deploy' : 'startInstance';
```

Comment:
Just for my understanding: This differs from [the way we track it in ET](https://docs.camunda.io/docs/components/modeler/desktop-modeler/telemetry/#deployment-event) (where we have all this inside the `deployment` context). I guess this is intended due to how Mixpanel works with the event names?

Reply:
Yes, we have a different naming strategy for Mixpanel